utility_network/display-utility-associations/src/main/java/com/esri/samples/display_utility_associations/DisplayUtilityAssociationsSample.java

```diff
@@ -0,0 +1,254 @@
+      });
+  
+      // create a legend
+      GridPane gridPane = new GridPane();
+      gridPane.getStyleClass().add("grid-pane");
+      gridPane.getColumnConstraints().addAll(Arrays.asList(new ColumnConstraints(25), new ColumnConstraints(100)));
+      gridPane.add(attachmentImageView, 0, 0);
+      gridPane.add(attachmentLabel, 1, 0);
+      gridPane.add(connectivityImageView, 0, 1);
+      gridPane.add(connectivityLabel, 1, 1);
+      
+      // add the map view and legend to the stack pane
+      stackPane.getChildren().addAll(mapView, gridPane);
+      StackPane.setAlignment(gridPane, Pos.TOP_LEFT);
+      StackPane.setMargin(gridPane, new Insets(10, 0, 0, 10));
+      
+    } catch (Exception e) {
+      // on any error, display the stack trace
+      e.printStackTrace();
+    }
+    
+  }
+  
+  /**
+   * Get utility association type for each association within extent , create graphics and add it to the graphics overlay.
+   */
+  private void addAssociationsGraphics() {
+    try {
+      // check if the current viewpoint is outside of the max scale
+      if (mapView.getCurrentViewpoint(Viewpoint.Type.CENTER_AND_SCALE).getTargetScale() >= 2000)
+        return;
+      
+      // check if the current viewpoint has an extent
+      Envelope extent = mapView.getCurrentViewpoint(Viewpoint.Type.BOUNDING_GEOMETRY).getTargetGeometry().getExtent();
+      if (extent != null) {
+        
+        // get all of the associations in the extent of the viewpoint
+        ListenableFuture<List<UtilityAssociation>> associationsFuture = utilityNetwork.getAssociationsAsync(extent);
+        associationsFuture.addDoneListener(() -> {
+          try {
+            List<UtilityAssociation> associations = associationsFuture.get();
+  
+            // iterate through all associations
+            associations.forEach(association -> {
+              
+              // check if the graphics overlay already contains the association
+              if (associationsOverlay.getGraphics().stream().noneMatch(graphic ->
+                graphic.getAttributes().containsKey("GlobalId")
+                  && graphic.getAttributes().get("GlobalId") == association.getGlobalId())
+                && association.getGeometry() != null) {
+  
+                // create attributes for the graphic so that the renderers display the appropriate symbol
+                Map<String, Object> attributes = Map.of(
+                  "GlobalId", association.getGlobalId(),
+                  "AssociationType", association.getAssociationType().toString());
+                
+                // create and add the graphic for the association
+                Graphic newGraphic = new Graphic(association.getGeometry(), attributes);
+                associationsOverlay.getGraphics().add(newGraphic);
+              }
+              
+            });
+          } catch (InterruptedException | ExecutionException e) {
```

Comment:
I noticed that it is this alert which fires when too many associations are in the extent: not the final catch. Let's recomment this catch, and add in some additional Alert logic since the error message that is returned is quite long. We can set the Error dialog window to be resized so a user could expand to see the error. 
```
          // this is thrown when there are too many associations in the extent
          } catch (InterruptedException | ExecutionException e) {
            Alert alert = new Alert(Alert.AlertType.ERROR, e.getMessage());
            alert.setResizable(true);
            alert.show();
```

(the message it shows is this "com.esri.arcgisruntime.ArcGISRuntimeException: Error getting associations: Maximum geometry count exceeded")